sdk/src/Services/S3/Generated/Model/Internal/MarshallTransformations/DeleteBucketAnalyticsConfigurationRequestMarshaller.cs

```diff
@@ -0,0 +1,101 @@
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ * 
+ *  http://aws.amazon.com/apache2.0
+ * 
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+/*
+ * Do not modify this file. This file is generated from the s3-2006-03-01.normal.json service model.
+ */
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using System.Text;
+using System.Xml.Serialization;
+
+using Amazon.S3.Model;
+using Amazon.Runtime;
+using Amazon.Runtime.Internal;
+using Amazon.Runtime.Internal.Transform;
+using Amazon.Runtime.Internal.Util;
+using System.Xml;
+
+#pragma warning disable CS0612,CS0618
+namespace Amazon.S3.Model.Internal.MarshallTransformations
+{
+    /// <summary>
+    /// DeleteBucketAnalyticsConfiguration Request Marshaller
+    /// </summary>       
+    public partial class DeleteBucketAnalyticsConfigurationRequestMarshaller : IMarshaller<IRequest, DeleteBucketAnalyticsConfigurationRequest> , IMarshaller<IRequest,AmazonWebServiceRequest>
+    {
+        /// <summary>
+        /// Marshaller the request object to the HTTP request.
+        /// </summary>  
+        /// <param name="input"></param>
+        /// <returns></returns>
+        public IRequest Marshall(AmazonWebServiceRequest input)
+        {
+            return this.Marshall((DeleteBucketAnalyticsConfigurationRequest)input);
+        }
+
+        /// <summary>
+        /// Marshaller the request object to the HTTP request.
+        /// </summary>  
+        /// <param name="publicRequest"></param>
+        /// <returns></returns>
+        public IRequest Marshall(DeleteBucketAnalyticsConfigurationRequest publicRequest)
+        {
+            var request = new DefaultRequest(publicRequest, "Amazon.S3");
+            PreMarshallCustomization(request, publicRequest);
+            request.HttpMethod = "DELETE";
+            request.AddSubResource("analytics");
+        
+            if (publicRequest.IsSetExpectedBucketOwner()) 
+            {
+                request.Headers["x-amz-expected-bucket-owner"] = publicRequest.ExpectedBucketOwner;
+            }
+            if (string.IsNullOrEmpty(publicRequest.BucketName))
+                throw new System.ArgumentException("BucketName is a required property and must be set before making this call.", "DeleteBucketAnalyticsConfigurationRequest.BucketName");
+            if (string.IsNullOrEmpty(publicRequest.AnalyticsId))
+                throw new AmazonS3Exception("Request object does not have required field AnalyticsId set");
+            
+            if (publicRequest.IsSetAnalyticsId())
+                request.Parameters.Add("id", StringUtils.FromString(publicRequest.AnalyticsId));
```

Comment:
This one used to be `AddSubResource` but was changed to a parameter (unlike `analytics` that remained the same).

I found a test so should be fine but wanted to confirm.